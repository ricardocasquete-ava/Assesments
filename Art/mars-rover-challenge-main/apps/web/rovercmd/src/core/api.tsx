import { baseUrl } from './apiConfig';

async function post<T> (url: string, data: Object): Promise<T> {
    const rawResponse = await fetch(`${baseUrl}${url}`, {
        method: 'POST',
        mode: 'cors',
        headers: {
            'Accept': 'application/json',
            'Content-Type': 'application/json'
        },
        body: JSON.stringify(data)
    });

    return rawResponse.json() as Promise<T>;
}

export { post };