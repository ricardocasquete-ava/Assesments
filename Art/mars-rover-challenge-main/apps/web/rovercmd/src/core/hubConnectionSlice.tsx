import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import type { RootState } from './store';

interface HubConnectionState {
    connectionId: string | null
};

const initialState: HubConnectionState = {
    connectionId: null
};

export const hubConnectionSlice = createSlice({
  name: 'hubConnection',
  initialState,
  reducers: {
    setConnectionId: (state, action: PayloadAction<string | null>) => {
      state.connectionId = action.payload
    }
  },
});

export const { setConnectionId } = hubConnectionSlice.actions;

export const getConnectionId = (state: RootState) => state.hubConnection.connectionId;

export default hubConnectionSlice.reducer;