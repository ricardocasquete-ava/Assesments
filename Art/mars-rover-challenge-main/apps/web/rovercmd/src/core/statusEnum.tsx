export enum Status {
    Ready = 0,
    Busy = 1,
    Error = -1
};