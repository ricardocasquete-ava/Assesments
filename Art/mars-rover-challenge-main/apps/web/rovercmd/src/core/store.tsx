import { configureStore } from '@reduxjs/toolkit';
import roverReducer from './roverSlice';
import hubConnectionReducer from './hubConnectionSlice';

const store = configureStore({
  reducer: {
    rover: roverReducer,
    hubConnection: hubConnectionReducer
  }
});

export default store;

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;