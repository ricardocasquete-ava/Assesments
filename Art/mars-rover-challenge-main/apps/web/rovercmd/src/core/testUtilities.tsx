import { render as reactRender, RenderResult } from '@testing-library/react'
import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'
import roverReducer from './roverSlice';
import hubConnectionReducer from './hubConnectionSlice';

function render(
  ui: React.ReactElement<any>,
  {
    preloadedState,
    store = configureStore({
      reducer: {
        rover: roverReducer,
        hubConnection: hubConnectionReducer
      },
      preloadedState
    }),
    ...renderOptions
  }: any = {}
): RenderResult {
  function Wrapper({ children }: any) {
    return <Provider store={store}>{children}</Provider>
  }
  return reactRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react'
// override render method
export { render }