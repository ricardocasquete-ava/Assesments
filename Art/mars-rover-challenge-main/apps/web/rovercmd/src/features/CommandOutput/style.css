.CommandOutput fieldset {
  border: none;
  padding: 0px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

.CommandOutput label {
  padding: 0px 10px;
}

.CommandOutput textarea {
  min-width: 250px;
}

.CommandOutput legend {
  padding: 10px 0px;
}
