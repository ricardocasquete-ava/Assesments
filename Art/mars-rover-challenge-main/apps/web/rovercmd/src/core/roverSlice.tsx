import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import type { AppDispatch, RootState } from './store';
import { postInput } from 'services/sendService';
import { Status } from './statusEnum';

interface RoverState {
    status: Status,
    output: string
};

const initialState: RoverState = {
    status: Status.Ready,
    output: ''
};

export const roverSlice = createSlice({
  name: 'rover',
  initialState,
  reducers: {
    setOutput: (state, action: PayloadAction<string>) => {
      state.output = action.payload
    },
    setStatus: (state, action: PayloadAction<Status>) => {
      state.status = action.payload
    }
  },
});

export const { setOutput , setStatus } = roverSlice.actions;

export const sendInput = (command: string) => (dispatch: AppDispatch) => {
  dispatch(setStatus(Status.Busy));
  dispatch(setOutput(''));

  postInput(command)
    .then(response => {
      dispatch(setOutput(response));
      dispatch(setStatus(Status.Ready));
    })
    .catch(error => {
      dispatch(setOutput(''));
      dispatch(setStatus(Status.Error));
    });
};

export const getOutput = (state: RootState) => state.rover.output;
export const getStatus = (state: RootState) => state.rover.status;

export default roverSlice.reducer;