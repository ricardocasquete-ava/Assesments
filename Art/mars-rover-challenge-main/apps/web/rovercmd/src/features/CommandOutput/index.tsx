import { useAppSelector } from 'core/hooks';
import { getOutput } from 'core/roverSlice';
import './style.css';

function CommandOutput() {
    const outputValue = useAppSelector(getOutput);

    return (
      <div className="CommandOutput">
        <fieldset>
          <legend>Output:</legend>
          <textarea
            rows={5}
            value={outputValue}
            readOnly />
        </fieldset>
      </div>
    );
  }

  export default CommandOutput;