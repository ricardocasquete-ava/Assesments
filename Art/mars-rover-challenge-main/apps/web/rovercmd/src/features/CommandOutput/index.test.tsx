import { render } from 'core/testUtilities';
import CommandOutput from './index';

test('renders CommandOutput', () => {
  const { container } = render(<CommandOutput />);
  expect(container.firstChild).toMatchSnapshot();
});
