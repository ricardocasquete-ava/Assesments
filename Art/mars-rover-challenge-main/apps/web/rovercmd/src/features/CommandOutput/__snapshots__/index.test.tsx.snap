// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CommandOutput 1`] = `
<div
  class="CommandOutput"
>
  <fieldset>
    <legend>
      Output:
    </legend>
    <textarea
      readonly=""
      rows="5"
    />
  </fieldset>
</div>
`;
